frontend/src/metabase/nav/components/TrialBanner/TrialBanner.tsx

```diff
@@ -0,0 +1,79 @@
+import dayjs from "dayjs";
+import { useMemo } from "react";
+import { msgid, ngettext, t } from "ttag";
+
+import { useUserSetting } from "metabase/common/hooks";
+import { Banner } from "metabase/components/Banner";
+import ExternalLink from "metabase/core/components/ExternalLink";
+import CS from "metabase/css/core/index.css";
+import { getStoreUrl } from "metabase/selectors/settings";
+import { Flex, Text } from "metabase/ui";
+import type { TokenStatus } from "metabase-types/api";
+export const TrialBanner = ({ tokenStatus }: { tokenStatus: TokenStatus }) => {
+  const [lastDismissed, setLastDismissed] = useUserSetting(
+    "trial-banner-dismissal-timestamp",
+  );
+
+  const now = dayjs();
+  const tokenExpiryDate = dayjs(tokenStatus["valid-thru"]);
+
+  const daysRemaining = tokenExpiryDate.diff(now, "day");
+  const lastDay = daysRemaining === 0;
+
+  const shouldShowBanner = useMemo(() => {
+    // No banner if the trial already expired
+    if (daysRemaining < 0) {
+      return false;
+    }
+
+    // In the last 3 days, check dismissal logic with stricter reappearance
+    if (daysRemaining <= 3) {
+      const bannerReappearanceThreshold = tokenExpiryDate
+        .subtract(daysRemaining, "day")
+        .unix();
+
+      // Banner shows if it hasn't been dismissed after the reappearance point
+      return !lastDismissed || lastDismissed < bannerReappearanceThreshold;
+    }
+
+    // For the first 11 days, check if it was ever dismissed
```

Comment:
Bug: there is no logic in the code that I can see that prevents this from triggering more than 11 days out. I'm pretty sure in its current state, every trial customer will see a notification that their trial will expire in X days, even if X is 90. The tests also don't cover any number of days greater than 11

Reply:
No bug, but very vague product documentation.
I double checked with the PM, and the verdict is - as soon as the trial starts (no matter how long it is), show the banner. I'll update tests to make this more explicit.

Reply:
🥴